monai/apps/reconstruction/mri_transforms.py

```diff
@@ -0,0 +1,301 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Sequence
+
+import numpy as np
+import torch
+from numpy import ndarray
+from torch import Tensor
+
+from monai.config.type_definitions import NdarrayOrTensor
+from monai.utils.type_conversion import convert_data_type, convert_to_tensor
+
+
+def convert_to_tensor_complex(data: ndarray) -> Tensor:
+    """
+    Convert numpy array to PyTorch tensor.
+    For complex arrays, the real and imaginary
+    parts are stacked along the last dimension.
+
+    Args:
+        data: Input numpy array
+
+    Returns:
+        PyTorch version of the data
+
+    Example:
+        .. code-block:: python
+
+            import numpy as np
+            data = np.array([ [1+1j, 1-1j], [2+2j, 2-2j] ])
+            # the following line prints (2,2)
+            print(data.shape)
+            # the following line prints torch.Size([2, 2, 2])
+            print(convert_to_tensor_complex(data).shape)
+    """
+    if np.iscomplexobj(data):
+        data = np.stack((data.real, data.imag), axis=-1)
+    return convert_data_type(data, torch.Tensor)[0]
+
+
+def complex_abs(x: NdarrayOrTensor) -> NdarrayOrTensor:
+    """
+    Compute the absolute value of a complex array.
+
+    Args:
+        x: Input array/tensor with 2 channels in the last dimension representing real and imaginary parts.
+
+    Returns:
+        Absolute value along the last dimention
+
+    Example:
+        .. code-block:: python
+
+            import numpy as np
+            x = np.array([3,4])[np.newaxis]
+            # the following line prints 5
+            print(complex_abs(x))
+    """
+    assert x.shape[-1] == 2
+    return (x[..., 0] ** 2 + x[..., 1] ** 2) ** 0.5
+
+
+class MaskFunc:
+    """
+    A basic class for under-sampling mask setup. It provides common features for under-sampling msak generators.
+    For example, RandomMaskFunc and EquispacedMaskFunc (two mask-generating objects defined right after this module)
+    both inherit MaskFunc to properly setup properties like the acceleration factor.
+    """
+
+    def __init__(self, center_fractions: Sequence[float], accelerations: Sequence[float]):
+        """
+        Args:
+            center_fractions: Fraction of low-frequency columns to be retained.
+                If multiple values are provided, then one of these numbers is chosen uniformly
+                each time.
+
+            accelerations: Amount of under-sampling. This should have the same length
+                as center_fractions. If multiple values are provided, then one of these is chosen
+                uniformly each time.
+        """
+        if len(center_fractions) != len(accelerations):
+            raise ValueError("Number of center fractions should match number of accelerations")
+
+        self.center_fractions = center_fractions
+        self.accelerations = accelerations
+        self.rng = np.random.RandomState()
```

Comment:
If inheriting from `Randomizable`, this line can be deleted, and `self.R` can be used instead.